src/test/java/org/broadinstitute/hellbender/utils/RandomDNA.java

```diff
@@ -0,0 +1,162 @@
+     */
+    public void nextBases(final byte[] destination) {
+        nextBases(destination, 0, destination.length);
+    }
+
+    /**
+     * Puts a random sequence of base into an byte array.
+     *
+     * @param dest   the destination array.
+     * @param offset first position on the destination array to fill.
+     * @param length how many based to put into the array.
+     * @throws IllegalArgumentException if {@code dest} is {@code null} or the pair {@code offset} and {@code length}
+     *                                  do not specify a valid index sub-interval in {@code dest}.
+     */
+    public void nextBases(final byte[] dest, final int offset, final int length) {
+        Utils.nonNull(dest, "input array cannot be null");
+        ParamUtils.inRange(offset, 0, dest.length, "the offset provided is not a valid index given the destination size");
+        ParamUtils.isPositiveOrZero(length, "the input length must be positive");
+        final int to = offset + length;
+        ParamUtils.inRange(to, 0, dest.length, "the length provided goes beyond the end of the destination array");
+        for (int i = offset; i < to; ) {
+            int randomInt = random.nextInt();
+            for (int j = 0; i < to && j < Integer.SIZE; j += 2) {
+                final int ord = randomInt & 0x000000003;
+                randomInt >>= 2;
+                switch (ord) {
+                    case 0:
+                        dest[i++] = 'A';
+                        break;
+                    case 1:
+                        dest[i++] = 'C';
+                        break;
+                    case 2:
+                        dest[i++] = 'G';
+                        break;
+                    case 3:
+                        dest[i++] = 'T';
+                        break;
+                    default:
+                        throw new GATKException.ShouldNeverReachHereException("this cannot be happening!!! " + ord);
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Returns a random DNA sequence of bases.
+     *
+     * @param size the length of the sequence.
+     * @return never {@code null}.
+     * @throws IllegalArgumentException if {@code size} is negative.
+     */
+    public byte[] nextBases(final int size) {
+        Utils.validateArg(size >= 0, "the size cannot be negative");
+        final byte[] result = new byte[size];
+        nextBases(result);
+        return result;
+    }
+
+    public File nextFasta(final SAMSequenceDictionary dict, final int basesPerLine)
+        throws IOException {
+        final File result = File.createTempFile("random-", ".fasta");
+        nextFasta(result, dict, basesPerLine);
+        return result;
+    }
+
+    public void nextFasta(final File out, final SAMSequenceDictionary dict, final int basesPerLine)
+            throws IOException {
+        final FileWriter writer = new FileWriter(out);
```

Comment:
Wrap in try() so that the close is done automatically?

Reply:
Done.